python/ccxt/bybit.py

```diff
@@ -5229,33 +5246,49 @@ def set_margin_mode(self, marginMode, symbol: Optional[str] = None, params={}):
     def change_margin_type(self, symbol, is_cross, leverage, is_long):
         self.load_markets()
         symbol = self.find_symbol(symbol)
-        return self.classify_change_margin(symbol, is_long, is_cross, leverage)
+        enableUnifiedMargin, enableUnifiedAccount = self.is_unified_enabled()
+        if enableUnifiedAccount and not self.is_inverse():
+            # TODO: get from account and avoid _change_margin_type/force_change_margin
+            if is_cross:
+                self.set_leverage(symbol, leverage=leverage)
+                return self._change_margin_type(is_cross)
+            else:
+                return self.classify_change_margin(symbol, is_long, is_cross, leverage, force_change_margin=True)
+        else:
+            return self.classify_change_margin(symbol, is_long, is_cross, leverage)
 
     @staticmethod
     def get_same_direction_position(positions, is_long):
+        no_side_position = None
         for position in positions:
             _is_long = position["is_long"]
-            if is_long == _is_long or _is_long is None or is_long is None:
+            if is_long == _is_long or is_long is None:
                 return position
 
-    def _change_margin_type(self, symbol, is_cross, leverage):
+            # making sure that we go through all position before we chose one without side.
+            if _is_long is None:
```

Comment:
you assume you have one which is okayish, but let's keep it the same behavior at least

```if _is_long is None and not no_side_position:```

Reply:
it's an assumption that was taken before, I will have to test it again but ok.